src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -58,14 +65,30 @@ public static class EnumerableMappingBuilder
         if (BuildElementMapping(ctx) is not { } elementMapping)
             return null;
 
+        // TODO Error: RMG008?
+        if (ctx.Target.IsArrayType()
+            || ctx.Target.ImplementsGeneric(ctx.Types.ImmutableArrayT, out _)
+            || ctx.Target.ImplementsGeneric(ctx.Types.ImmutableListT, out _))
+        {
+            return null;
+        }
+
+        if (ctx.Target.ImplementsGeneric(ctx.Types.List, out _))
```

Comment:
Is a `Select` + `AddRange` mapping faster than a `foreach` + `Add` mapping (assumed https://github.com/riok/mapperly/issues/251 is implemented)? It makes sense to me if the `elementMapping` is synthetic, but does it also make sense for non-synthetic mappings?